segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -421,6 +436,10 @@ private Void handleException(long requestId, String segment, String operation, T
             int containerId = ((ContainerNotFoundException) u).getContainerId();
             log.warn("Wrong host. Segment = '{}' (Container {}) is not owned. Operation = '{}').", segment, containerId, operation);
             connection.send(new WrongHost(requestId, segment, ""));
+        } else if ( u instanceof ReadCancellationException) {
+            log.info("Closing connection {} while reading segment {} due to CancellationException.", connection, segment);
+            connection.send(new SegmentRead(segment, requestId, true, false, EMPTY_BYTE_BUFFER));
+            connection.close();
```

Comment:
I don't think closing the connection is actually needed with this reply added.

Reply:
yes, connection.close() is not required. Removed it.